LoadNGem compiler error on MSVC Debug build

### Expected behavior
The `DataHandling` target should build on MSVC in Debug

### Actual behavior
It does not:
```
Build FAILED.

"mantid-build\Framework\DataHandling\DataHandling.vcxproj" (default target) (1) ->
(ClCompile target) ->
 mantid\Framework\DataHandling\src\LoadNGEM.cpp(114,1): error C3016: 'i': index variable in OpenMP 'for' statement must have signed integral type [mantid-build\Framework\DataHandling\DataHandling.vcxproj]
```

It looks like OpenMP is complaining about being passed an unsigned `size_t` integer. Replacing with an `int` or `long` as required fixes this.
https://github.com/mantidproject/mantid/blob/1cfc6edf2c9562e7c50d7fc3f7257101ace02809/Framework/DataHandling/src/LoadNGEM.cpp#L115

### Steps to reproduce the behavior
Try to build the `.\Framework\DataHandling\DataHandling.vcxproj` target in debug mode on Windows

### Platforms affected
Windows

--- a/Framework/DataHandling/src/LoadNGEM.cpp
+++ b/Framework/DataHandling/src/LoadNGEM.cpp
@@ -112,11 +112,10 @@ void createEventWorkspace(const int &maxToF, const double &binWidth,
   dataWorkspace = DataObjects::create<DataObjects::EventWorkspace>(
       NUM_OF_SPECTRA, HistogramData::Histogram(HistogramData::BinEdges(xAxis)));
   PARALLEL_FOR_NO_WSP_CHECK()
-  for (size_t i = 0; i < events.size(); ++i) {
+  for (int i = 0; i < NUM_OF_SPECTRA; ++i) {
     dataWorkspace->getSpectrum(i) = events[i];
-    const int index = static_cast<int>(i + 1);
-    dataWorkspace->getSpectrum(i).setSpectrumNo(index);
-    dataWorkspace->getSpectrum(i).setDetectorID(index);
+    dataWorkspace->getSpectrum(i).setSpectrumNo(i + 1);
+    dataWorkspace->getSpectrum(i).setDetectorID(i + 1);
   }
   dataWorkspace->setAllX(HistogramData::BinEdges{xAxis});
   dataWorkspace->getAxis(0)->unit() =

